app/services/check_answers/section_lister_service.rb

```diff
@@ -28,13 +28,15 @@ def call(session_data)
               Sections::Dependants.new(check),
               Sections::ClientIncome.new(check),
             ]
-            post = [Sections::Outgoings.new(check)]
+            post = check.gross_early_eligibility_exit? ? [] : [Sections::Outgoings.new(check)]
             finance_sections = if check.partner?
                                  pre + [Sections::PartnerIncome.new(check)] + post
```

Comment:
I think we also have to put in a ternary operator here, for the partner income i.e

`pre + (check.gross_early_eligibility_exit? ? [] : [Sections::PartnerIncome.new(check)]) + post`

We are showing these screens even though user did not fill those questions in.

Reply:
thanks good spot, I think I was just expecting to see income questions.